pkg/state/state_cni_plugins.go

```diff
@@ -146,13 +146,17 @@ func (s *stateCNIPlugins) GetManifestObjects(
 	if staticConfig == nil {
 		return nil, errors.New("staticConfig provider required")
 	}
+	clusterInfo := catalog.GetClusterTypeProvider()
+	if clusterInfo == nil {
+		return nil, errors.New("clusterInfo provider required")
+	}
 	renderData := &CNIPluginsManifestRenderData{
 		CrSpec:       cr.Spec.SecondaryNetwork.CniPlugins,
 		Tolerations:  cr.Spec.Tolerations,
 		NodeAffinity: cr.Spec.NodeAffinity,
 		RuntimeSpec: &cniRuntimeSpec{
 			runtimeSpec:        runtimeSpec{config.FromEnv().State.NetworkOperatorResourceNamespace},
-			CniBinDirectory:    utils.GetCniBinDirectory(staticConfig, nil),
+			CniBinDirectory:    utils.GetCniBinDirectory(staticConfig, clusterInfo),
```

Comment:
Was there a bug before that ?

Reply:
Not really, as CNI plugins are not supposed to be used in OpenShift.
It would choose CNI default path instead of OpenShift one.